src/BlockTypes/RelatedProducts.php

```diff
@@ -75,16 +83,82 @@ public function update_query( $pre_render, $parsed_block ) {
 	 */
 	public function build_query( $query ) {
 		$parsed_block = $this->parsed_block;
-		if ( ! ProductQuery::is_woocommerce_variation( $parsed_block ) && 'woocommerce/related-products' !== $parsed_block['attrs']['namespace'] ) {
+		if ( ! $this->is_related_products_block( $parsed_block ) ) {
 			return $query;
 		}
-		$product          = wc_get_product();
-		$related_products = wc_get_related_products( $product->get_id() );
+
+		$related_products_ids = $this->get_related_products_ids();
+		if ( count( $related_products_ids ) < 1 ) {
+			return array();
+		}
 
 		return array(
 			'post_type'   => 'product',
-			'post__in'    => $related_products,
+			'post__in'    => $related_products_ids,
 			'post_status' => 'publish',
 		);
 	}
+
+	/**
+	 * If there are no related products, return an empty string.
+	 *
+	 * @param string $content The block content.
+	 * @param array  $block The block.
+	 *
+	 * @return string The block content.
+	 */
+	public function render_block( string $content, array $block ) {
+		if ( ! $this->is_related_products_block( $block ) ) {
```

Comment:
I'm wondering if this conditional is really necessary, because by default on line 120, it will always return the $content, except if the `count( $related_products_ids ) < 1` (on line 117)

Reply:
This function is hooked to the `render_block` filter. So this means that every function hooked to the `render_block` is executed for each block that will be rendered. For this reason, I do this check.